diff --git CMakeLists.txt CMakeLists.txt
index 3d38291..fd3811a 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -8,6 +8,8 @@ if(NOT CATKIN_DISABLED)
   find_package(catkin QUIET)
 endif()
 
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIC")
+
 find_package(console_bridge REQUIRED)
 
 if(${catkin_FOUND})
diff --git cmake/PocoConfig.cmake cmake/PocoConfig.cmake
index 727d826..6ea1d13 100644
--- cmake/PocoConfig.cmake
+++ cmake/PocoConfig.cmake
@@ -161,9 +161,13 @@ IF(Poco_INCLUDE_DIR)
     ENDIF(Poco_LIBRARY_DIR AND Poco_FOUNDATION_LIB)
     IF(Poco_FOUNDATION_LIB)
       IF ("${CMAKE_BUILD_TYPE}" STREQUAL "Debug")
-	SET(DBG "d")
+        IF (APPLE)
+          SET(DBG "")
+        ELSE (APPLE)
+          SET(DBG "d")
+        ENDIF (APPLE)
       ELSE ("${CMAKE_BUILD_TYPE}" STREQUAL "Debug")
-	SET(DBG "")
+	      SET(DBG "")
       ENDIF ("${CMAKE_BUILD_TYPE}" STREQUAL "Debug")
       SET(Comp_List "Foundation${DBG}")
       FOREACH(COMPONENT ${Poco_FIND_COMPONENTS})
