--- console_bridge/CMakeLists.txt
+++ console_bridge/CMakeLists.txt
@@ -33,8 +33,19 @@ endif()
 add_library(${PROJECT_NAME} src/console.cpp)
 target_link_libraries(${PROJECT_NAME} ${Boost_LIBRARIES})
 
-install(TARGETS ${PROJECT_NAME}
-  DESTINATION lib/)
+if(WIN32)
+  set(LIB_INSTALL_DIR bin) # .dll are in PATH, like executables
+else(WIN32)
+  set(LIB_INSTALL_DIR lib)
+endif(WIN32)
+
+set(_INSTALL_DESTINATIONS
+	RUNTIME DESTINATION bin
+	LIBRARY DESTINATION ${LIB_INSTALL_DIR}
+	ARCHIVE DESTINATION lib
+)
+
+install(TARGETS ${PROJECT_NAME} ${_INSTALL_DESTINATIONS})
 
 install(DIRECTORY include/
   DESTINATION include
