diff --git a/CMakeLists.txt b/CMakeLists.txt
index ea4f46a..cf2611d 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -7,6 +7,9 @@ set_property(GLOBAL PROPERTY PREDEFINED_TARGETS_FOLDER "CMake")
 
 project(cpr)
 
+find_package(lib_manager)
+lib_defaults()
+
 if(NOT ${CMAKE_VERSION} LESS 3.2)
     set(CMAKE_CXX_STANDARD 11)
     set(CMAKE_CXX_STANDARD_REQUIRED ON)
@@ -56,8 +59,8 @@ endmacro()
 message(STATUS "C++ Requests CMake Options")
 message(STATUS "=======================================================")
 cpr_option(USE_SYSTEM_CURL
-    "If ON, this project will look in the system paths for an installed curl library" OFF)
-cpr_option(BUILD_CPR_TESTS "Set to ON to build cpr tests." ON)
+    "If ON, this project will look in the system paths for an installed curl library" ON)
+cpr_option(BUILD_CPR_TESTS "Set to ON to build cpr tests." OFF)
 cpr_option(GENERATE_COVERAGE "Set to ON to generate coverage reports." OFF)
 cpr_option(CPR_CURL_NOSIGNAL "Set to ON to disable use of signals in libcurl." OFF)
 cpr_option(USE_SYSTEM_GTEST
diff --git a/cpr/CMakeLists.txt b/cpr/CMakeLists.txt
index 39c6c12..cb83637 100644
--- cpr/CMakeLists.txt
+++ cpr/CMakeLists.txt
@@ -43,6 +43,22 @@ message(STATUS "Using CURL_LIBRARIES: ${CURL_LIBRARIES}.")
 target_link_libraries(${CPR_LIBRARIES}
     ${CURL_LIBRARIES})
 
+file(GLOB public_headers ${CPR_INCLUDE_DIRS}/cpr/[a-z]*.h)
+
+install(
+	FILES ${public_headers}
+	DESTINATION include/cpr
+)
+
+install(TARGETS ${CPR_LIBRARIES}
+	LIBRARY DESTINATION lib
+	ARCHIVE DESTINATION lib
+)
+
+set(PC_FILE ${CMAKE_BINARY_DIR}/cpr.pc)
+configure_file("cpr.pc.in" ${PC_FILE} @ONLY)
+install(FILES ${PC_FILE} DESTINATION lib/pkgconfig)
+
 if(NOT (CMAKE_VERSION VERSION_LESS 3.0))
     target_include_directories(${CPR_LIBRARIES}
         PUBLIC
--- cpr/cpr.pc.in	1970-01-01 01:00:00.000000000 +0100
+++ cpr/cpr.pc.in	2017-04-04 12:33:25.000000000 +0200
@@ -0,0 +1,12 @@
+prefix=@CMAKE_INSTALL_PREFIX@
+exec_prefix=@CMAKE_INSTALL_PREFIX@
+libdir=${prefix}/lib
+includedir=${prefix}/include/
+
+Name: minizip
+Description: Minizip
+Version: @PROJECT_VERSION@
+Libs: -L${libdir} -lcpr -lcurl
+
+Cflags: -I${includedir}
+
