diff --git a/CMakeLists.txt b/CMakeLists.txt
index 75a8255..fc7c7fd 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -7,9 +7,9 @@ project(inja LANGUAGES CXX VERSION 3.4.0)
 option(INJA_USE_EMBEDDED_JSON "Use the shipped json header if not available on the system" ON)
 option(INJA_INSTALL "Generate install targets for inja" ON)
 option(INJA_EXPORT "Export the current build tree to the package registry" ON)
-option(BUILD_TESTING "Build unit tests" ON)
-option(INJA_BUILD_TESTS "Build unit tests when BUILD_TESTING is enabled." ON)
-option(BUILD_BENCHMARK "Build benchmark" ON)
+option(BUILD_TESTING "Build unit tests" OFF)
+option(INJA_BUILD_TESTS "Build unit tests when BUILD_TESTING is enabled." OFF)
+option(BUILD_BENCHMARK "Build benchmark" OFF)
 option(COVERALLS "Generate coveralls data" OFF)
 
 
