diff --git CMakeLists.txt CMakeLists.txt
index 6839a29..5d9cc0e 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -1,6 +1,6 @@
 # CMakeLists.txt has to be located in the project folder and cmake has to be
 # executed from 'project/build' with 'cmake ../'.
-cmake_minimum_required(VERSION 3.0)
+cmake_minimum_required(VERSION 3.3)
 find_package(Rock)
 project(lib_config VERSION 0.1)
 rock_init()
diff --git bindings/python/CMakeLists.txt bindings/python/CMakeLists.txt
index 5b07cb4..511d47c 100644
--- bindings/python/CMakeLists.txt
+++ bindings/python/CMakeLists.txt
@@ -2,11 +2,11 @@ FIND_PACKAGE(PythonInterp REQUIRED)
 FIND_PACKAGE(PythonLibs REQUIRED)
 
 if (PYTHON_VERSION_MAJOR EQUAL 3)
-    FIND_PACKAGE(Boost COMPONENTS python${PYTHON_VERSION_MAJOR}${PYTHON_VERSION_MINOR})
-    if(NOT DEFINED Boost_python${PYTHON_VERSION_MAJOR}${PYTHON_VERSION_MINOR}_FOUND)
-        message("Boost python${PYTHON_VERSION_MAJOR}${PYTHON_VERSION_MINOR} not found. Trying Boost python${PYTHON_VERSION_MAJOR}")
-        FIND_PACKAGE(Boost COMPONENTS python${PYTHON_VERSION_MAJOR} REQUIRED)
-    endif()
+    FIND_PACKAGE(Boost COMPONENTS python)#${PYTHON_VERSION_MAJOR}.${PYTHON_VERSION_MINOR})
+    # if(NOT DEFINED Boost_python${PYTHON_VERSION_MAJOR}${PYTHON_VERSION_MINOR}_FOUND)
+    #     message("Boost python${PYTHON_VERSION_MAJOR}${PYTHON_VERSION_MINOR} not found. Trying Boost python${PYTHON_VERSION_MAJOR}")
+    #     FIND_PACKAGE(Boost COMPONENTS python${PYTHON_VERSION_MAJOR} REQUIRED)
+    # endif()
 endif()
 
 set(PYTHON_INSTALL_PATH ${CMAKE_INSTALL_PREFIX}/lib/python${PYTHON_VERSION_MAJOR}.${PYTHON_VERSION_MINOR}/site-packages)
