--- include/ode/common.h	Mon Jun  2 23:09:37 2008
+++ include/ode/common.h	Wed Aug  8 08:43:51 2012
@@ -517,6 +517,7 @@
   dVector3 t1;		/* torque applied to body 1 */
   dVector3 f2;		/* force applied to body 2 */
   dVector3 t2;		/* torque applied to body 2 */
+  dReal lambda;
 } dJointFeedback;
 
 
--- ode/src/step.cpp	Thu Jul 24 22:18:40 2008
+++ ode/src/step.cpp	Mon Oct 15 12:58:57 2012
@@ -461,7 +461,7 @@
     dJointFeedback *fb/*=NULL*/, unsigned jointBodyIndex)
 {
     dIASSERT (infom > 0 && cforce && JRow && rowRhsLambda);
-    dReal sumLX = 0, sumLY = 0, sumLZ = 0, sumAX=0, sumAY = 0, sumAZ = 0;
+    dReal sumLX = 0, sumLY = 0, sumLZ = 0, sumAX=0, sumAY = 0, sumAZ = 0, sumLambda = 0;
     const dReal *currJ = JRow, *currLambda = rowRhsLambda + RLE_LAMBDA;
     for (unsigned int k = infom; ; ) {
         const dReal lambda = *currLambda;
@@ -475,10 +475,12 @@
             break;
         }
         currJ += JME__MAX;
+        sumLambda += lambda;
         currLambda += RLE__RHS_LAMBDA_MAX;
     }
     if (fb != NULL) {
-        if (jointBodyIndex == dJCB__MIN) {
+      fb->lambda = sumLambda;
+      if (jointBodyIndex == dJCB__MIN) {
             fb->f1[dV3E_X] = sumLX;
             fb->f1[dV3E_Y] = sumLY;
             fb->f1[dV3E_Z] = sumLZ;
