diff --git a/CMakeLists.txt b/CMakeLists.txt
index 70a9edc..24c6cef 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -11,6 +11,13 @@
   ${sisl_SOURCE_DIR}/include
   )
 
+if(WIN32)
+       set(CMAKE_SHARED_LINKER_FLAGS "-Wl,--enable-auto-import")
+       set(CMAKE_MODULE_LINKER_FLAGS "-Wl,--enable-auto-import")
+else()
+       set(CMAKE_CXX_FLAGS "-fPIC")
+       set(CMAKE_C_FLAGS "-fPIC")
+endif()
 
 # Make the sisl library
 
@@ -33,10 +40,10 @@
   SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wno-unused-but-set-variable -fPIC")
   SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -Wno-unused-but-set-variable -fPIC")
 ENDIF(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
-IF(WIN32)
-  SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /MP8 -D_SCL_SECURE_NO_WARNINGS -D_CRT_SECURE_NO_WARNINGS")
-  SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} /MP8 -D_SCL_SECURE_NO_WARNINGS -D_CRT_SECURE_NO_WARNINGS")
-ENDIF(WIN32)
+# IF(WIN32)
+#   SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /MP8 -D_SCL_SECURE_NO_WARNINGS -D_CRT_SECURE_NO_WARNINGS")
+#   SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} /MP8 -D_SCL_SECURE_NO_WARNINGS -D_CRT_SECURE_NO_WARNINGS")
+# ENDIF(WIN32)
 
 
 # Apps, examples, tests, ...?
@@ -129,18 +136,18 @@
 
 # 'install' target
 
-IF(WIN32)
-  # Windows
-  SET(CMAKE_INSTALL_PREFIX CACHE INTERNAL "")
-  SET(sisl_INSTALL_PREFIX "$ENV{PROGRAMFILES}/SINTEF/sisl"
-    CACHE PATH "Path to install SISL")
-  INSTALL(TARGETS sisl DESTINATION ${sisl_INSTALL_PREFIX}/lib)
-  INSTALL(DIRECTORY include/
-    DESTINATION ${sisl_INSTALL_PREFIX}/include
-    FILES_MATCHING PATTERN "*.h"
-    PATTERN ".svn" EXCLUDE
-    )
-ELSE(WIN32)
+# IF(WIN32)
+#   # Windows
+#   SET(CMAKE_INSTALL_PREFIX CACHE INTERNAL "")
+#   SET(sisl_INSTALL_PREFIX "$ENV{PROGRAMFILES}/SINTEF/sisl"
+#     CACHE PATH "Path to install SISL")
+#   INSTALL(TARGETS sisl DESTINATION ${sisl_INSTALL_PREFIX}/lib)
+#   INSTALL(DIRECTORY include/
+#     DESTINATION ${sisl_INSTALL_PREFIX}/include
+#     FILES_MATCHING PATTERN "*.h"
+#     PATTERN ".svn" EXCLUDE
+#     )
+# ELSE(WIN32)
   # Linux
   # lib
   INSTALL(TARGETS sisl DESTINATION lib COMPONENT sisl)
@@ -151,8 +158,7 @@
     FILES_MATCHING PATTERN "*.h"
     PATTERN ".svn" EXCLUDE
     )
-ENDIF(WIN32)
-
+#ENDIF(WIN32)
 
 # CPack stuff
 IF(NOT GoTools_ALL_COMPILE)
