diff --git a/manifest.xml b/manifest.xml
index bbcaea4..c1fa01b 100644
--- manifest.xml
+++ manifest.xml
@@ -17,7 +17,7 @@
     <depend package="base/logging" />
 
     <depend package="rice" optional="1" />
-    <depend package="external/sisl" optional="1" />
+    <depend package="external/sisl" />
     <depend package="gui/vizkit3d" optional="1" />
     <tags>stable</tags>
 </package>
diff --git a/src/JointLimitRange.hpp b/src/JointLimitRange.hpp
index 63ce05a..0f98b16 100644
--- src/JointLimitRange.hpp
+++ src/JointLimitRange.hpp
@@ -2,6 +2,7 @@
 #define BASE_JOINT_LIMIT_RANGE_HPP
 
 #include <base/JointState.hpp>
+#include <string>
 
 namespace base
 {
diff --git a/src/Time.cpp b/src/Time.cpp
index c4fa634..81e6834 100644
--- src/Time.cpp
+++ src/Time.cpp
@@ -296,7 +296,13 @@ Time Time::fromString(const string& stringTime, Time::Resolution resolution,
     }
 
     struct tm tm;
+ /* TODO: windows does not support strptime, so this whole function
+          is unsusable on windows! */
+#ifdef WIN32
+	    throw std::runtime_error("base::Time::fromString failed- not implemented for windows");
+#else
     if (NULL == strptime(mainTime.c_str(), mainFormat.c_str(), &tm))
+#endif
     {
         throw runtime_error(
             "Time::fromString failed: " + mainTime + "' did not match the given "
diff --git a/src/Time.hpp b/src/Time.hpp
index 93382f3..4ed27e2 100644
--- src/Time.hpp
+++ src/Time.hpp
@@ -6,6 +6,11 @@
 #include <stdint.h>
 #include <vector>
 
+#ifdef WIN32
+#include <Time.h>
+typedef long suseconds_t;
+#endif
+
 namespace base
 {
     struct Time
diff --git a/viz/TrajectoryVisualization.hpp b/viz/TrajectoryVisualization.hpp
index de2edbe..eabb22e 100644
--- viz/TrajectoryVisualization.hpp
+++ viz/TrajectoryVisualization.hpp
@@ -1,10 +1,10 @@
 #ifndef TRAJECTORYVISUALISATION_H
 #define TRAJECTORYVISUALISATION_H
+#include <vizkit3d/Vizkit3DPlugin.hpp>
 #include <Eigen/Geometry>
 #include <osg/Geometry>
 #include <osg/LineWidth>
 #include <deque>
-#include <vizkit3d/Vizkit3DPlugin.hpp>
 #include <base/geometry/Spline.hpp>
 #include <base/Trajectory.hpp>
 
