--- CMakeLists.txt
+++ CMakeLists.txt
@@ -29,7 +29,7 @@ endif()
 # This shouldn't be necessary, but there has been trouble
 # with MSVC being set off, but MSVCXX ON.
 if(MSVC OR MSVC90 OR MSVC10)
-  set(MSVC ON)
+  set(MSVC OFF)
 endif (MSVC OR MSVC90 OR MSVC10)
 
 set(CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
@@ -58,11 +58,11 @@ enable_testing()
 
 add_subdirectory(urdf_parser)
 
-if(WIN32 AND NOT CYGWIN)
-  set(CMAKE_CONFIG_INSTALL_DIR CMake)
-else()
+#if(WIN32 AND NOT CYGWIN)
+#  set(CMAKE_CONFIG_INSTALL_DIR CMake)
+#else()
   set(CMAKE_CONFIG_INSTALL_DIR ${CMAKE_INSTALL_DATAROOTDIR}/${PROJECT_NAME}/cmake/)
-endif()
+#endif()
 
 set(PKG_NAME ${PROJECT_NAME})
 set(PKG_LIBRARIES urdfdom_sensor urdfdom_model_state urdfdom_model urdfdom_world)
--- urdf_parser/CMakeLists.txt
+++ urdf_parser/CMakeLists.txt
@@ -33,10 +33,23 @@ add_executable(urdf_unit_test test/urdf_unit_test.cpp)
 target_link_libraries(urdf_unit_test urdfdom_model ${Boost_LIBRARIES})
 add_test(urdf_unit_test urdf_unit_test)
 
-INSTALL(TARGETS urdfdom_model DESTINATION ${CMAKE_INSTALL_LIBDIR})
-INSTALL(TARGETS urdfdom_world DESTINATION ${CMAKE_INSTALL_LIBDIR})
+
+if(WIN32)
+  set(LIB_INSTALL_DIR bin) # .dll are in PATH, like executables
+else(WIN32)
+  set(LIB_INSTALL_DIR lib)
+endif(WIN32)
+
+set(_INSTALL_DESTINATIONS
+	RUNTIME DESTINATION bin
+	LIBRARY DESTINATION ${LIB_INSTALL_DIR}
+	ARCHIVE DESTINATION lib
+)
+
+INSTALL(TARGETS urdfdom_model ${_INSTALL_DESTINATIONS})
+INSTALL(TARGETS urdfdom_world ${_INSTALL_DESTINATIONS})
 INSTALL(TARGETS check_urdf urdf_to_graphiz urdf_mem_test
-  DESTINATION ${CMAKE_INSTALL_BINDIR})
-INSTALL(TARGETS urdfdom_sensor DESTINATION ${CMAKE_INSTALL_LIBDIR})
-INSTALL(TARGETS urdfdom_model_state DESTINATION ${CMAKE_INSTALL_LIBDIR})
-INSTALL(DIRECTORY include/urdf_parser DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
+   ${_INSTALL_DESTINATIONS})
+INSTALL(TARGETS urdfdom_sensor ${_INSTALL_DESTINATIONS})
+INSTALL(TARGETS urdfdom_model_state ${_INSTALL_DESTINATIONS})
+INSTALL(DIRECTORY include/urdf_parser  DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
--- urdf_parser/include/urdf_parser/exportdecl.h
+++ urdf_parser/include/urdf_parser/exportdecl.h
@@ -47,7 +47,7 @@
 //
 // On Linux, set the visibility accordingly. If C++ symbol visibility
 // is handled by the compiler, see: http://gcc.gnu.org/wiki/Visibility
-# if defined _WIN32 || defined __CYGWIN__
+# if defined _WIN_32 || defined __CYGWIN__
 // On Microsoft Windows, use dllimport and dllexport to tag symbols.
 #  define URDFDOM_DLLIMPORT __declspec(dllimport)
 #  define URDFDOM_DLLEXPORT __declspec(dllexport)
