diff --git CMakeLists.txt CMakeLists.txt
index e554f7c..9377b99 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -31,7 +31,7 @@ endif()
 # This shouldn't be necessary, but there has been trouble
 # with MSVC being set off, but MSVCXX ON.
 if(MSVC OR MSVC90 OR MSVC10)
-  set(MSVC ON)
+  set(MSVC OFF)
 endif (MSVC OR MSVC90 OR MSVC10)
 
 set(CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
@@ -62,11 +62,11 @@ set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/${CMAKE_INSTALL_LIBDIR}"
 
 enable_testing()
 
-if(WIN32 AND NOT CYGWIN)
-  set(CMAKE_CONFIG_INSTALL_DIR CMake)
-else()
-  set(CMAKE_CONFIG_INSTALL_DIR ${CMAKE_INSTALL_LIBDIR}/${PROJECT_NAME}/cmake)
-endif()
+#if(WIN32 AND NOT CYGWIN)
+#  set(CMAKE_CONFIG_INSTALL_DIR CMake)
+#else()
+set(CMAKE_CONFIG_INSTALL_DIR ${CMAKE_INSTALL_LIBDIR}/${PROJECT_NAME}/cmake)
+#endif()
 string(REGEX REPLACE "[^/]+" ".." RELATIVE_PATH_CMAKE_DIR_TO_PREFIX "${CMAKE_CONFIG_INSTALL_DIR}")
 string(REGEX REPLACE "[^/]+" ".." RELATIVE_PATH_LIBDIR_TO_PREFIX "${CMAKE_INSTALL_LIBDIR}")
 
diff --git urdf_parser/CMakeLists.txt urdf_parser/CMakeLists.txt
index 6efc955..3ccc7cd 100644
--- urdf_parser/CMakeLists.txt
+++ urdf_parser/CMakeLists.txt
@@ -89,20 +89,32 @@ target_include_directories(urdf_mem_test PUBLIC include)
 target_link_libraries(urdf_mem_test urdfdom_model)
 
 include(CTest)
-if(BUILD_TESTING)
-  # TODO: check Shane's comment https://github.com/ros/urdfdom/pull/157/files#r664960227
-  add_subdirectory(test)
+if(NOT WIN32)
+  if(BUILD_TESTING)
+    # TODO: check Shane's comment https://github.com/ros/urdfdom/pull/157/files#r664960227
+    add_subdirectory(test)
+  endif()
 endif()
 
+if(WIN32)
+  set(LIB_INSTALL_DIR bin) # .dll are in PATH, like executables
+else(WIN32)
+  set(LIB_INSTALL_DIR lib)
+endif(WIN32)
+
+set(_INSTALL_DESTINATIONS
+	RUNTIME DESTINATION bin
+	LIBRARY DESTINATION ${LIB_INSTALL_DIR}
+	ARCHIVE DESTINATION lib
+)
+
 INSTALL(
   TARGETS
   check_urdf
   urdf_to_graphiz
   urdf_to_graphviz
   urdf_mem_test
-  RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
-  LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
-  ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+  ${_INSTALL_DESTINATIONS}
 )
 INSTALL(
   TARGETS
@@ -112,9 +124,7 @@ INSTALL(
   urdfdom_model_state
   EXPORT
   urdfdom
-  RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
-  LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
-  ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+  ${_INSTALL_DESTINATIONS}
 )
 install(
   EXPORT urdfdom
diff --git urdf_parser/include/urdf_parser/exportdecl.h urdf_parser/include/urdf_parser/exportdecl.h
index e01e783..654dcad 100644
--- urdf_parser/include/urdf_parser/exportdecl.h
+++ urdf_parser/include/urdf_parser/exportdecl.h
@@ -47,7 +47,7 @@
 //
 // On Linux, set the visibility accordingly. If C++ symbol visibility
 // is handled by the compiler, see: http://gcc.gnu.org/wiki/Visibility
-# if defined _WIN32 || defined __CYGWIN__
+# if defined _WIN_32 || defined __CYGWIN__
 // On Microsoft Windows, use dllimport and dllexport to tag symbols.
 #  define URDFDOM_DLLIMPORT __declspec(dllimport)
 #  define URDFDOM_DLLEXPORT __declspec(dllexport)
